fairscale/nn/pipe/async_schedule.py

```diff
@@ -11,15 +11,64 @@
 
 import torch
 from torch import Tensor, nn
+from torch.autograd.profiler import record_function
 from torch.distributed import ProcessGroup
 
 from fairscale.nn.model_parallel import get_pipeline_parallel_ranks
 
+from .checkpoint import Checkpointing
 from .messages import Transport
 from .microbatch import Batch
-from .multiprocess_pipeline import create_task
-from .skip.tracker import SkipTrackerThroughPotals
-from .types import EVENT_LOOP_QUEUE, PipeMessage, Tensors
+from .skip.tracker import SkipTrackerThroughPotals, use_skip_tracker
+from .types import EVENT_LOOP_QUEUE, PipeMessage, TensorOrTensors, Tensors
+from .worker import Task
+
+
+def create_task(
+    checkpoint_stop: int,
+    i: int,
```

Comment:
nit: can we rename these to what state they are tracking (part_id and chunk_id I think)?